src/Device/jprogOperations.js

```diff
@@ -138,20 +140,27 @@ const verifySerialPortAvailable = device => {
  * @returns {Promise} Promise that resolves if successful or rejects with error.
  */
 async function validateFirmware(device, fwVersion) {
+    let valid = false;
     try {
         const fwInfo = await nrfDeviceLib.readFwInfo(
-            getDeviceLibContext(),
+            deviceLibContext,
             device.id
         );
-        const valid = fwInfo.imageInfoList.find(imageInfo =>
-            imageInfo.version.string.includes(fwVersion)
-        );
+        if (
+            typeof fwVersion === 'object' &&
+            typeof fwVersion.validator === 'function'
+        ) {
+            valid = fwVersion.validator(fwInfo.imageInfoList);
```

Comment:
This one is overwritten by the next line after the if block.

Reply:
Oops, good catch, there was supposed to be an else there, wonder if auto-formatting changed that somehow.